cpp/src/dictionary/search.cu

```diff
@@ -151,12 +143,20 @@ struct find_insert_index_fn {
             std::enable_if_t<std::is_same<Element, dictionary32>::value or
                              std::is_same<Element, list_view>::value or
                              std::is_same<Element, struct_view>::value>* = nullptr>
-  std::unique_ptr<scalar> operator()(dictionary_column_view const& input,
-                                     scalar const& key,
-                                     rmm::cuda_stream_view stream,
-                                     rmm::mr::device_memory_resource* mr) const
+  std::unique_ptr<scalar> operator()(dictionary_column_view const&,
+                                     scalar const&,
+                                     rmm::cuda_stream_view,
+                                     rmm::mr::device_memory_resource*) const
   {
-    CUDF_FAIL("column cannot be the keys for dictionary");
+    if constexpr (std::is_same_v<Element, dictionary32>) {
+      CUDF_FAIL("dictionary column cannot be the keys for a dictionary");
+    }
+    if constexpr (std::is_same_v<Element, list_view>) {
+      CUDF_FAIL("list_view column cannot be the keys for a dictionary");
+    }
+    if constexpr (std::is_same_v<Element, struct_view>) {
+      CUDF_FAIL("struct_view column cannot be the keys for a dictionary");
+    }
```

Comment:
These separate messages are really unnecessary. I wrote this code so I'm confident that a single message is fine here.

Reply:
Awesome, I will yank them.

Reply:
Removed in latest version.